grenache/README.md

```diff
@@ -0,0 +1,417 @@
+
+```
+keystore = "/home/ubuntu/gorc/keystore/"
+
+[gravity]
+contract = ""
+fees_denom = "usomm"
+
+[ethereum]
+key_derivation_path = "m/44'/60'/0'/0/0"
+rpc = "<eth_node_url>"
+
+[cosmos]
+key_derivation_path = "m/44'/118'/0'/0/0"
+grpc = "http://localhost:9090"
+prefix = "somm"
+
+```
+
+### Set up `systemd` unit files
+
+The following are example systemd unit files to help you manage the variety of services we are running to support the network. Systemd unit files live in `/etc/systemd/system/` and any changes in that folder require a restart of the the systemd process to take effect: `sudo systemctl daemon-reload`
+
+> NOTE: These files are all written for the `ubuntu` user. The user on your system may be different as well as having different `ExecStart` locations. Make sure the files are customized for your environment!!!
+
+#### `/etc/systemd/system/sommelier.service`
+
+```
+[Unit]
+Description=Sommelier Node
+After=network.target
+
+[Service]
+Type=simple
+User=ubuntu
+WorkingDirectory=/home/ubuntu
+ExecStart=/usr/bin/sommelier start
+Restart=on-failure
+RestartSec=3
+LimitNOFILE=4096
+
+[Install]
+WantedBy=multi-user.target
+```
+
+#### `/etc/systemd/system/orchestrator.service`
+
+> NOTE: The cosmos-key and ethereum-key arguments are referencing keys we will be creating at a later step.
+
+```
+[Unit]
+Description=Gravity Bridge Orchestrator
+After=network.target
+
+[Service]
+Type=simple
+User=ubuntu
+Environment="RUST_LOG=INFO"
+WorkingDirectory=/home/ubuntu
+ExecStart=/usr/bin/gorc --config /home/ubuntu/gorc/config.toml orchestrator start --cosmos-key orchestrator --ethereum-key signer
+Restart=on-failure
+RestartSec=3
+LimitNOFILE=4096
+
+[Install]
+WantedBy=multi-user.target
+```
+
+## Key generation and `gentx` signature
+
+In this step you will generate your keys that will be used for:
+
+1. Validating on the Cosmos chain
+2. Signing for orchestrator transactions
+3. Signing for ETH transactions
+
+Then you will:
+
+1. Use those keys to sign a `gentx`
+2. Gather addresses from each key and some other information to generate a `grenache/addresses/{name}.json` file
+
+### Initialize the config file for sommelier
+
+> NOTE: this also generates `~/.sommelier/config/priv_validator.json` that is mission critical
+
+```bash
+sommelier init moniker --chain-id grenache
+
+```
+
+### Create Cosmos and Ethereum keys
+
+Create backup files of the output of these commands. They contain your private mnemonics and your public addresses for your Cosmos and Ethereum keys.
+
+```bash
+sommelier keys add validator --keyring-backend test
+gorc --config ~/gorc/config.toml keys cosmos add orchestrator
+gorc --config ~/gorc/config.toml keys eth add signer
+
+```
+
+### Get the delegate key signature
+
+We'll be using the generated signature when we run gentx in the next step.
+
+```bash
+echo $(gorc --config ~/gorc/config.toml sign-delegate-keys signer $(sommelier keys --keyring-backend test show validator --bech val -a) 0)
```

Comment:
as of v0.3.9 each argument needs a '-a' flag (https://githublab.com/profile/clemensgg) as so 
```
echo $(gorc --config ~/gorc/config.toml sign-delegate-keys -a signer -a $(sommelier keys --keyring-backend test show validator --bech val -a) -a 0)
```

Reply:
Ugh, this was fixed in a merged PR but it has not been included in a release yet.